nixos/immich-machine-learning: huggingface-hub permissions error

### Nixpkgs version

- Unstable (25.11)

### Describe the bug

The `immich-machine-learning` service runs into a permissions error on my system. I'm not sure what the root cause is, but the error itself seems to come from `_download_to_tmp_and_move` in `.../python3.12/site-packages/huggingface_hub/file_download.py` (see logs).

### Steps to reproduce

My config can be found at https://code.tecosaur.net/tec/golgi/src/branch/master/modules/immich.nix.

In order to trigger the error I went to the Immich Web UI, attempted to search my images for a term, and then after observing an error message looked in the journal to find the permissions error.

### Expected behaviour

Successful model downloading and running.

### Screenshots

_No response_

### Relevant log output

```console
[2025-06-21 16:12:56 +0000] [3495] [INFO] Starting gunicorn 23.0.0
[2025-06-21 16:12:56 +0000] [3495] [INFO] Listening at: http://127.0.0.1:3003 (3495)
[2025-06-21 16:12:56 +0000] [3495] [INFO] Using worker: immich_ml.config.CustomUvicornWorker
[2025-06-21 16:12:56 +0000] [3612] [INFO] Booting worker with pid: 3612
mkdir -p failed for path /var/empty/.config/matplotlib: [Errno 1] Operation not permitted: '/var/empty/.config'
Matplotlib created a temporary cache directory at /tmp/matplotlib-tutvnd0b because there was an issue with the default path (/var/empty/.config/matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable dire>
[2025-06-21 16:12:57 +0000] [3612] [INFO] Started server process [3612]
[2025-06-21 16:12:57 +0000] [3612] [INFO] Waiting for application startup.
[2025-06-21 16:12:57 +0000] [3612] [INFO] Application startup complete.
[274B blob data]
[2025-06-21 16:34:32 +0000] [3612] [ERROR] Exception in ASGI application
 21 16:34:32 nucleus machine-learning[3612]:   File "/nix/store/bd6mggnl9i9i0cisbmxnz71z8mkzsfdk-python3.12-uvicorn-0.34.0/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]]: Traceback (most recent call last):

...

  File "/nix/store/gzlsxhxc2xniqj64l96kpxc3yy775j98-python3.12-fastapi-0.115.12/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/gzlsxhxc2xniqj64l96kpxc3yy775j98-python3.12-fastapi-0.115.12/lib/python3.12/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/74y4qajd72cmqzh5mwciwnhl6rdid85i-immich-machine-learning-1.134.0/lib/python3.12/site-packages/immich_ml/main.py", line 177, in predict
    response = await run_inference(inputs, entries)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

...

  File "/nix/store/gn26zyfipl6js5dyczyyilkkmwl5pinb-python3.12-huggingface-hub-0.32.3/lib/python3.12/site-packages/huggingface_hub/file_download.py", line 988, in hf_hub_download
    return _hf_hub_download_to_local_dir(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/gn26zyfipl6js5dyczyyilkkmwl5pinb-python3.12-huggingface-hub-0.32.3/lib/python3.12/site-packages/huggingface_hub/file_download.py", line 1290, in _hf_hub_download_to_local_dir
    _download_to_tmp_and_move(
  File "/nix/store/gn26zyfipl6js5dyczyyilkkmwl5pinb-python3.12-huggingface-hub-0.32.3/lib/python3.12/site-packages/huggingface_hub/file_download.py", line 1710, in _download_to_tmp_and_move
    xet_get(
  File "/nix/store/gn26zyfipl6js5dyczyyilkkmwl5pinb-python3.12-huggingface-hub-0.32.3/lib/python3.12/site-packages/huggingface_hub/file_download.py", line 627, in xet_get
    download_files(
RuntimeError: Data processing error: I/O error: Operation not permitted (os error 1)
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.15.2, NixOS, 25.11 (Xantusia), 25.11pre-git`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.3`
 - nixpkgs: `/nix/store/xnr2xjs0clkhlhqfc3vqcmhnnwdk3c5p-source`

### Notify maintainers

@jvanbruegge

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+nixos%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/nixos/modules/services/web-apps/immich.nix
+++ b/nixos/modules/services/web-apps/immich.nix
@@ -360,6 +360,7 @@ in
       MACHINE_LEARNING_WORKERS = "1";
       MACHINE_LEARNING_WORKER_TIMEOUT = "120";
       MACHINE_LEARNING_CACHE_FOLDER = "/var/cache/immich";
+      XDG_CACHE_HOME = "/var/cache/immich";
       IMMICH_HOST = "localhost";
       IMMICH_PORT = "3003";
     };

